packages/back-end/src/app.ts

```diff
@@ -400,6 +400,22 @@ app.post(
   "/subscription/new-pro-trial",
   stripeController.postNewProTrialSubscription
 );
+app.post(
+  "/subscription/payment-methods/setup-intent",
+  stripeController.postSetupIntent
+);
+app.get(
+  "/subscription/payment-methods/:subscriptionId",
+  stripeController.listPaymentMethods
+);
+app.post(
+  "/subscription/payment-methods/detach",
+  stripeController.deletePaymentMethod
+);
+app.post(
+  "/subscription/payment-methods/set-default",
+  stripeController.postStripeCustomerDefaultCard
+);
```

Comment:
These new endpoints only work on cloud.  They should be wrapped in an `IS_CLOUD` check so they aren't even added to the app when self-hosting.  Same with the front-end code.